### ISSUE TITLE
returning 1 or -1 to custom_stoploss actually resets the stoploss to 100%

### ISSUE THREAD

**Issue by macsim26** [freqtrade/freqtrade#10046]
## Describe your environment

  * Operating system: ____Ubuntu 22.04
  * Python Version: _____ 3.10
  * CCXT version: _____ 4.2.67
  * Freqtrade Version: ____2024.2 (update to 2024.3 done but is marked as 2024.2...)
  

*Explain the problem you have encountered*
Using custom_stoploss as indicated in the help by returning 1 or -1 to make no change push really the stoploss to -100%
### Steps to reproduce:

  1.def custom_stoploss(self, pair: str, trade: 'Trade', current_time: datetime, current_rate: float, current_profit: float, **kwargs) -> float:
    
    sl_new = 1

    if (current_time - timedelta(minutes=15) >= trade.open_date_utc):

        dataframe, _ = self.dp.get_analyzed_dataframe(pair, self.timeframe)
        current_candle = dataframe.iloc[-1].squeeze()
        current_profit = trade.calc_profit_ratio(current_candle['close'])

        if (current_profit >= 0.5):
            sl_new = 0.25

    return sl_new

![Capture du 2024-04-05 12-03-31](https://github.com/freqtrade/freqtrade/assets/88237677/d8fd0c6c-f651-443e-a077-e422a38d7b14)

  2.def custom_stoploss(self, pair: str, trade: 'Trade', current_time: datetime, current_rate: float, current_profit: float, **kwargs) -> float:
    
    sl_new = None

    if (current_time - timedelta(minutes=15) >= trade.open_date_utc):

        dataframe, _ = self.dp.get_analyzed_dataframe(pair, self.timeframe)
        current_candle = dataframe.iloc[-1].squeeze()
        current_profit = trade.calc_profit_ratio(current_candle['close'])

        if (current_profit >= 0.5):
            sl_new = 0.25

    return sl_new

![Capture du 2024-04-05 12-04-06](https://github.com/freqtrade/freqtrade/assets/88237677/9e065422-429a-457d-b770-1b614232adec)

  * What happened? stoploss is reinitialised to 100%
  * What did you expect to happen? stoploss should have been keeped as the doc says?

Thanks.
Miguel

**Comment by froggleston** [comment]
What's your initial strategy `stoploss` value? If it's -0.99 or -1 or something similar, that will always be your default. Stoploss *cannot* be set lower than the existing stoploss.

From the docs, you can see that custom_stoploss returns a relative stoploss to the *current* price. If you want to use an absolute stoploss value in custom_stoploss, use stoploss_from_open or one of the other associated helper functions: https://www.freqtrade.io/en/stable/strategy-callbacks/#common-helpers-for-stoploss-calculations

**Comment by macsim26** [comment]
Hi initial stoploss was -0.25 and the two codes should have normally the same result no?
As you told me hundreds times i read the doc again and again like a bible...
Can i have a refund if the doc put me at loss? ;) It's a joke "no pain no gain"... Thanks a lot for freqtrade!!!

**Comment by macsim26** [comment]
![Capture du 2024-04-05 12-41-00](https://github.com/freqtrade/freqtrade/assets/88237677/f96e9880-2419-44ed-a0e5-dbd0ba7b5cbe)
Dreamed of a new car will only get good shoes...

**Comment by xmatthias** [comment]
can you please share your full custom-stoploss function? (including header and everything).
Please use a code-block for that (tripple backticks (`)).
thanks.

**Comment by macsim26** [comment]
Hello here is the code i use : 
```

from freqtrade.strategy.interface import IStrategy
from typing import Dict, List, Optional
from functools import reduce
from pandas import DataFrame

import talib.abstract as ta
import numpy as np
import pandas as pd
import freqtrade.vendor.qtpylib.indicators as qtpylib
import datetime
import requests
from technical.util import resample_to_interval, resampled_merge
from datetime import datetime, timedelta, timezone
from freqtrade.persistence import Trade
from freqtrade.strategy import stoploss_from_open, merge_informative_pair, DecimalParameter, IntParameter, CategoricalParameter
import technical.indicators as ftt

class bybit4(IStrategy):

    INTERFACE_VERSION = 3

    position_adjustment_enable = False

    # ROI table:
    minimal_roi = {
        "0": 100,
    }

    # Stoploss:
    stoploss = -0.3
    # Trailing stop:
    trailing_stop = False
    trailing_stop_positive = 0.005
    trailing_stop_positive_offset = 0.03
    trailing_only_offset_is_reached = False

    # Sell signal
    use_exit_signal = True
    exit_profit_only = False
    exit_profit_offset = 0.267
    ignore_roi_if_entry_signal = False

    # Optimal timeframe for the strategy
    timeframe = '1m'

    process_only_new_candles = True
    startup_candle_count = 4320
    use_custom_stoploss = True

    def custom_stoploss(self, pair: str, trade: 'Trade', current_time: datetime, current_rate: float, current_profit: float, **kwargs) -> float:
    
        sl_new = None

        if (current_time - timedelta(minutes=1) >= trade.open_date_utc):

            dataframe, _ = self.dp.get_analyzed_dataframe(pair, self.timeframe)
            current_candle = dataframe.iloc[-1].squeeze()
            current_profit = trade.calc_profit_ratio(current_candle['close'])

            if (current_profit >= 0.45):
                sl_new = 0.2

        return sl_new
```
Thanks

**Comment by xmatthias** [comment]
> Using custom_stoploss as indicated in the help by returning 1 or -1 to make no change push really the stoploss to -100

well first of - this is not true (not sure where you read this - but it's not in [the docs](https://www.freqtrade.io/en/stable/strategy-callbacks/#custom-stoploss)).

a "no action" return value should be "None" - not 1 / -1 - and all examples on the above linked page use None - not -1.. 

A stoploss of 1 (100%) would be a 10% price move on a 10x position ... so there can be reasons to try and risk more than 100% of the initial capital (though it's not something i'd personally recommend / use - but that's besides the point).

---

I think this will need a full, minimal reproduction (strategy, pairs, timerange) if you do think this is a problem. 
In my tests, behavior between 1 and None is identical - with stoploss never exceeding 30%.

**Comment by macsim26** [comment]
> well first of - this is not true (not sure where you read this - but it's not in [the docs](https://www.freqtrade.io/en/stable/strategy-callbacks/#custom-stoploss)).

Thanks for your answer when i talk about -1 in the doc i made reference to this part wich was the code i used in first instance :
https://www.freqtrade.io/en/stable/strategy-callbacks/#trailing-stoploss-with-positive-offset

For me everything is ok now using None so i close this issue...

Have you noticed freqtrade 2024.03 announce to be 2024.02 or my updates are broken?

Thanks a lot...
Miguel

**Comment by xmatthias** [comment]
nice catch
i looked over it 3 times and didn't spot any with -1 :laughing: 


> Have you noticed freqtrade 2024.03 announce to be 2024.02 or my updates are broken?

Don't see how that'd be happening.
i can also run `pip install freqtrade==2024.3 ` - and it'll report 2024.3 properly.

**Comment by macsim26** [comment]
> Don't see how that'd be happening.

ok my requirements.txt was modified to use last CCXT blocking update...

Thanks a lot...
